src/app/service-offering/service-offering-selector/service-offering-selector.component.ts

```diff
@@ -37,46 +44,6 @@ export class ServiceOfferingSelectorComponent implements ControlValueAccessor {
     this.change = new EventEmitter();
   }
 
-  @Input()
-  public get serviceOffering(): ServiceOffering {
-    return this._serviceOffering;
-  }
-
-  public set serviceOffering(serviceOffering: ServiceOffering) {
-    this._serviceOffering = serviceOffering;
-    this.propagateChange(this.serviceOffering);
-  }
-
-  public get customOfferingDescription(): Observable<string> {
-    if (!this.serviceOffering || !this.serviceOffering.iscustomized || !areCustomParamsSet(this.serviceOffering)) {
-      return Observable.of('');
-    }
-
-    const cpuNumber = this.serviceOffering.cpunumber;
-    const cpuSpeed = this.serviceOffering.cpuspeed;
-    const memory = this.serviceOffering.memory;
-
-    return this.translateService.get(['UNITS.MB', 'UNITS.MHZ'])
-      .map(translations => {
-        return `${cpuNumber}x${cpuSpeed} ${translations['UNITS.MHZ']}, ${memory} ${translations['UNITS.MB']}`;
-      });
-  }
-
-  public changeOffering(change: MatSelectChange): void {
-    const newOffering = this.serviceOfferings.find(_ => _.id === change.value);
-    this.saveOffering();
-    this.serviceOffering = newOffering;
-    if (newOffering.iscustomized) {
-      this.showCustomOfferingDialog()
-        .subscribe(customOffering => {
-          this.setCustomOffering(customOffering);
-          this.change.next(this.serviceOffering);
-        });
-    } else {
-      this.change.next(this.serviceOffering);
-    }
-  }
-
   public registerOnChange(fn): void {
```

Comment:
Do we need these methods of `ControlValueAccessor` if you got rid of implementation?